src/state/routing/utils.ts

```diff
@@ -89,6 +116,17 @@ const parsePair = ({ reserve0, reserve1 }: V2PoolInRoute): Pair =>
     CurrencyAmount.fromRawAmount(parseToken(reserve1.token), reserve1.quotient)
   )
 
-function isV3Route(route: V3PoolInRoute[] | V2PoolInRoute[]): route is V3PoolInRoute[] {
-  return route[0].type === 'v3-pool'
+function isVersionedRoute<T extends V2PoolInRoute | V3PoolInRoute>(
+  type: T['type'],
+  route: (V3PoolInRoute | V2PoolInRoute)[]
+): route is T[] {
+  return route.every((pool) => pool.type === type)
+}
+
+export function currencyAddressForSwapQuote(currency: Currency): string {
```

Comment:
naming fn w/r/t its consumer instead of its functionality feels wrong to me

Reply:
could rename to `currencyAddressForRoutingApi` but technically this should just be used for getting swap quotes - will also be not a thing once i ask the routing team to support `NATIVE` as an exported constant